### PR TITLE
added basic functionality

### PR DESCRIPTION
**Pull request by nbykh**
* `Utils.hpp`: functions to compare double values
* `Point.hpp`: class `Point` for a point on a plane, i.e. pair of (x,y) coordinates
* `Edge.hpp`: a pair of points
* `Triangle.hpp`: class `Triangle` i.e. a triple of edges
* `Plgs.hpp`: stores an instance of a planar straight-line graph

### CODE REVIEW

**Comment by nbykh** [comment]
to be merged after #8

**Comment by nbykh** [comment]
> to be merged after #8

the dependency got resolved as #8 requires additional effort

File: src/Edge.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+Edge::Edge(const Point &pointA, const Point &pointB)
+    : _pointA{pointA}, _pointB{pointB},
+      _length{Point::distance(pointA, pointB)} {}
+
+bool Edge::isCongruent(const Edge &other) const {
+  return isEqual(_length, other._length);
+}
+
+bool Edge::isOpposite(const Edge &other) const {
```
Comment (nbykh) [inline src/Edge.cpp:14]: this functionality seems redundant for now

File: src/Edge.cpp [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  if (!isEqual(_length, other._length)) {
+    return isLess(_length, other._length);
+  }
+  if (!(_pointA == other._pointA)) {
+    return _pointA < other._pointA;
+  }
+  return _pointB < other._pointB;
+}
+
+Edge Edge::operator!() const { return {_pointB, _pointA}; }
```
Comment (nbykh) [inline src/Edge.cpp:32]: remove this functionality for now

File: src/Edge.hpp [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  [[nodiscard]] bool isOpposite(const Edge &other) const;
+
+  [[nodiscard]] bool operator==(const Edge &other) const;
+
+  [[nodiscard]] bool operator<(const Edge &other) const;
+
+  [[nodiscard]] Edge operator!() const;
+
+private:
+  Point _pointA{};
```
Comment (nbykh) [inline src/Edge.hpp:26]: better to store points in a set s.t. edge is uniqely defined

File: src/Point.hpp [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  [[nodiscard]] static double distance(const Point &pointA,
+                                       const Point &pointB);
+
+private:
+  double _x{0.0};
+  double _y{0.0};
+};
+} // namespace triangulator
+
+template <> struct std::hash<triangulator::Point> {
```
Comment (nbykh) [inline src/Point.hpp:30]: redundant for now

File: src/Plgs.cpp [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+namespace triangulator {
+
+Pslg::Pslg(const nlohmann::json &json) {
+  _instanceUid = json["instance_uid"];
+  const auto numPoints = json["num_points"].get<size_t>();
+  if (json["points_x"].size() != numPoints ||
+      json["points_y"].size() != numPoints) {
+    throw std::logic_error("invalid json");
+  }
+  for (auto iPoint = 0U; iPoint < numPoints; ++iPoint) {
```
[the remark this answers is not in the document]
Comment (nbykh) [reply src/Plgs.cpp:14]: usually it's handy to define local variables by auto (their type is defined from the right-hand side of the expression). `0U` is a literal for `unsigned`, so it's similar to `unsigned iPoint = 0U;`

File: src/Triangle.hpp [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+#define _TRIANGLE_HPP_
+
+#include "Edge.hpp"
+#include "Point.hpp"
+#include <set>
+
+namespace triangulator {
+class Triangle {
+public:
+  Triangle(const std::set<Point> &corners);
```
[the remark this answers is not in the document]
Comment (nbykh) [reply src/Triangle.hpp]: fixed

File: src/Plgs.cpp (hunk 5 above)
[the remark this answers is not in the document]
Comment (Damir21) [reply src/Plgs.cpp:14]: Yes, but why we need to use unsigned?

[the remark this answers is not in the document]
Comment (nbykh) [reply src/Plgs.cpp:14]: I thought there was no much difference, but now I'm thinking that using `size_t` is more convenient for iteration over containers. I'll fix it

[the remark this answers is not in the document]
Comment (nbykh) [reply src/Plgs.cpp]: fixed

**Review by Damir21 — APPROVED** [review]

**Pull request merged**